src/polymer/dom-calls-to-native.ts

```diff
@@ -0,0 +1,119 @@
+/**
+ * @license
+ * Copyright (c) 2018 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import babelTraverse from 'babel-traverse';
+import * as babel from 'babel-types';
+import {Document, isPositionInsideRange, ParsedDocument, Severity, SourceRange, Warning} from 'polymer-analyzer';
+
+import {registry} from '../registry';
+import {Rule} from '../rule';
+import {stripIndentation, stripWhitespace} from '../util';
+
+
+class DomCallsToNative extends Rule {
+  code = 'dom-calls-to-native';
+  description = stripIndentation(`
+      Warns when the Polymer.dom is used in places where native
+      methods can now be used instead.
+
+        Polymer.dom(event).path
+
+      Accepted syntax:
+
+        event.composedPath()
+  `);
+
+  private _replacements: Map<string, string> = new Map([
+    ['localTarget', 'target'],
+    ['rootTarget', 'composedPath()[0]'],
+    ['path', 'composedPath()']
+  ]);
+
+  async check(document: Document) {
+    const warnings: Warning[] = [];
+
+    const docs = document.getFeatures({kind: 'js-document'});
+
+    for (const doc of docs) {
+      babelTraverse(doc.parsedDocument.ast, {
+        noScope: true,
+        MemberExpression: (path) => {
+          if (!babel.isIdentifier(path.node.property)) {
+            return;
+          }
+
+          const name = path.node.property.name;
+
+          if (!this._isPolymerDomCall(path.node.object) ||
+              !this._replacements.has(name)) {
+            return;
+          }
+
+          const containingDoc =
+              getParsedDocumentContaining(doc.sourceRange, document);
+
+          if (!containingDoc) {
+            return;
+          }
+
+          const sourceRange = containingDoc.sourceRangeForNode(path.node)!;
+          const replacement = this._replacements.get(name)!;
```

Comment:
The ! is a code smell. Consider:

```typescript
const replacement = this._replacements.get(name);

 if (!this._isPolymerDomCall(path.node.object) || replacement === undefined) {
  return;
}
```

and similarly early exiting if sourceRange is undefined (I know there's no current way for that source range to be undefined, but I've seen enough weird things over the past couple years)